### ISSUE TITLE
Ability to stage individual lines in a more discoverable way

### ISSUE THREAD

**Issue by Aligorith** [sublimehq/sublime_merge#465]
**Problem description**

Sometimes, Git decides to show multiple unrelated changes together as a single hunk, or perhaps you'd like to have more control about splitting out several sets of unrelated but co-developed changes when committing them.

Right now it is possible to stage individual lines by selecting them first, and then hitting the "Stage Lines" button that appears below the normal "Stage Hunk". However, I only discovered that this functionality existed at all while starting to write up this issue after spending a week wishing it did, and only after the Github related-issue search recommended Issue #206 to me.

TLDR: It's hard to discover that it's possible to stage/unstage individual lines

**Preferred solution**
To make it more discoverable, it would be awesome if hovering over the line numbers shows a button/widget over the line number that, when clicked, would stage/unstage the offending line.

**Alternatives**
If it's not feasible to implement or too distracting, I could live with an entry in a right click menu that appears when clicking on part of the offending line. That's what Git GUI had (i.e. the TK/TCL tool I've been using for many years until last week, when I started using SublimeMerge as I didn't have install permissions on my work box;  BTW, I'm really pleased happy that this tool exists in the first place - being forced to use the command line only was like developing with one hand tied behind my back while juggling knives!)

**Comment by kakra** [comment]
I agree... Not being able to commit a selection of lines instead of a complete hunk is one of the points stopping me from buying a licence. But after your hint, I see that the button at the hunk context actually changes to "stage lines". Nice! Thanks.

Actually, I'd also wish it was more discoverable. I was right clicking on the lines, so maybe a context menu item should be added. It could also add a small button "stage selection" to a bunch of selected lines. Single-line staging is not really needed on my side (I could simply just select part of the line instead) but I like your idea. Of course, the same features are needed on the unstage side.

I vote for this, +1.

**Comment by Birch-san** [comment]
+1 I came searching because I couldn't find the functionality. my brother even went back to using SourceTree because he believed Sublime Merge didn't support staging lines.
